src/sentry/api/serializers/models/team.py

```diff
@@ -38,16 +39,30 @@ def get_attrs(self, item_list, user):
         else:
             access_requests = frozenset()
 
+        if user.is_authenticated():
+            # map of org id to role
+            org_roles = {
+                om.organization_id: om.role for om in
+                OrganizationMember.objects.filter(
+                    user=user,
+                    organization__in=set([t.organization_id for t in item_list]),
+                )}
+        else:
+            org_roles = {}
+
         is_superuser = (request and is_active_superuser(request) and request.user == user)
         result = {}
         for team in item_list:
             is_member = team.id in memberships
+            org_role = org_roles.get(team.organization_id)
             if is_member:
                 has_access = True
             elif is_superuser:
                 has_access = True
             elif team.organization.flags.allow_joinleave:
                 has_access = True
+            elif org_role and roles.get(org_role).is_global:
```

Comment:
maybe add a comment here about what is_global means? nit.

Reply:
i'll add this separately. i don't wanna wait for the build to pass again for a comment